### DISCUSSION TITLE
Build app for old Android app

### DISCUSSION THREAD

**Discussion by faustopala** [apache/cordova#495] [Q&A]
Hello guys, it’s possible to continue to buold an android app witch cordova 7.0.0 for Android 4.4 now that JCenter was shut down? I cannot find some dependency in Maven Central. How can i override default location in gradle to download from local file?

thank you

**Comment by breautek** [comment]
Short answer is no, not really.

Gradle builds does support multiple repositories and will search each repo for a requested dependency. This would involve manipulating the generated cordova project (I believe the `settings.gradle` file). So if you can find an existing mirror that mirrored the old jcenter repositories and contains the required dependencies... then that could be your golden ticket.

Otherwise you would have to source those dependencies manually which would be an immense amount of work. Dependency trees tends to be massive so sourcing those dependencies manually is something not very realistic. If any part of the dependency tree depends on closed-source software, it might be even impossible to manually reconstruct the dependency tree.

But assuming that all the missing dependencies can be obtained, you would probably then have to create your own maven repository (which could be local server) and add that to the repos list in the gradle build config. This way the dependencies can be resolved by the gradle build, falling back to your own local maven server for the otherwise missing dependencies.
